certloader/listener.go

```diff
@@ -29,14 +29,13 @@ type Listener struct {
 	net.Listener
 
 	cert   Certificate
-	config *tls.Config
+	config TLSServerConfig
 }
 
-func NewListener(listener net.Listener, cert Certificate, config *tls.Config) *Listener {
+func NewListener(listener net.Listener, config TLSServerConfig) *Listener {
```

Comment:
I like that we're collapsing certificate and config, makes the code cleaner for sure.

Reply:
👍